### DISCUSSION TITLE
Writing a .i file to study the impact of temperature gradient on grain growth in a bicrystal system

### DISCUSSION THREAD

**Discussion by rakeshm22** [idaholab/moose#29627] [Q&A Tools]
Hello everyone,
I am trying to reproduce grain growth for a bicrystal system in the presence of a temperature gradient as an external field. I wrote a .i file based on the equations mentioned in the paper.
I am not able to find any output. Kindly help in this regards.

![Screenshot from 2024-12-31 11-20-19](https://github.com/user-attachments/assets/4a57c736-5f92-459e-8636-66aa73966203)

(https://github.com/user-attachments/files/18279534/Michael.R_2013.pdf)

[Mesh]
type = GeneratedMesh
dim = 2
nx = 20
ny = 10
xmin = 0
xmax = 20
ymin = 0
ymax = 10
[]

[GlobalParams]
block = 0
[]

[Variables]
[./eta1]
[../]

[./eta2]
[../]

[./T]
order = FIRST
family = LAGRANGE
[../]
[]

[ICs]

[./IC_eta1]
x1 = 0
y1 = 0
x2 = 20
y2 = 10
inside = 1.0
outside = 0.0
type = BoundingBoxIC
variable = eta1
int_width = 0.5
[../]

[./IC_eta2]
x1 = 10
y1 = 0
x2 = 20
y2 = 10
inside = 1.0
outside = 0.0
type = BoundingBoxIC
variable = eta2
int_width = 0.5
[../]

[./T]
x1 = 0
y1 = 0
x2 = 18
y2 = 10
inside = 0.01
outside = 0.01
type = BoundingBoxIC
variable = T
int_width = 0.5
[../]

[]

[AuxVariables]
[./T_x]
order = CONSTANT
family = MONOMIAL
[../]
[./T_y]
order = CONSTANT
family = MONOMIAL
[../]

[./eta1_x]
order = CONSTANT
family = MONOMIAL
[../]
[./eta1_y]
order = CONSTANT
family = MONOMIAL
[../]

[./eta2_x]
order = CONSTANT
family = MONOMIAL
[../]
[./eta2_y]
order = CONSTANT
family = MONOMIAL
[../]

[]

[AuxKernels]
[./T_x]
type = VariableGradientComponent
variable = T_x
component = x
gradient_variable = T
[../]
[./T_y]
type = VariableGradientComponent
variable = T_y
component = y
gradient_variable = T
[../]
[./eta1_x]
type =VariableGradientComponent
variable = eta1_x
component = x
gradient_variable = eta1
[../]

[./eta1_y]
type =VariableGradientComponent
variable = eta1_y
component = y
gradient_variable = eta1
[../]

[./eta2_x]
type =VariableGradientComponent
variable = eta2_x
component = x
gradient_variable = eta2
[../]

[./eta2_y]
type =VariableGradientComponent
variable = eta2_y
component = y
gradient_variable = eta2
[../]

[]

[Kernels]

[./AC1_bulk]
type = AllenCahn
variable = eta1
f_name = F
args = 'eta2 T '
[../]
[./AC1_int]
type = ACInterface
variable = eta1
kappa_name = kappa_s
[../]
[./e1_dot]
type = TimeDerivative
variable = eta1
[../]

[./AC2_bulk]
type = AllenCahn
variable = eta2
f_name = F
args = 'eta1 T '
[../]
[./AC2_int]
type = ACInterface
variable = eta2
kappa_name = kappa_s
[../]
[./e2_dot]
type = TimeDerivative
variable = eta2
[../]

[./diff2]
type = Diffusion
variable = T
[../]

[]

[Materials]

[./FreeEng]
type = DerivativeParsedMaterial

args = 'eta1 eta2 T_x T_y eta1_x eta1_y eta2_x eta2_y  T'

f_name = F

constant_names = ' ww'  
constant_expressions = ' 1.14 '

function = 'f2:=(1/4eta1^4-1/2eta1^2+1/4eta2^4-1/2eta2^2);
f3:=s*(eta1^2eta2^2);
f4:=eta14.0*(T_xeta1_x+T_yeta1_y);
f5:=eta24.0(T_xeta2_x+T_yeta2_y);

        F:= ww*(f2 + f3 + 0.25) + f4 + f5;F'

derivative_order = 2
outputs = exodus

[../]

[./const]
type = GenericConstantMaterial
prop_names = 'kappa_s L '
prop_values = '1.0 1.0 '
outputs = exodus
[../]

[]

[BCs]

[./Periodic]
[./all]
auto_direction = 'y'
[../]
[../]

[./neumaan1]
type = NeumannBC
boundary = 'left right'
variable = eta1
value = 0
[../]
[./neumaan2]
type = NeumannBC
boundary = 'left right'
variable = eta2
value = 0
[../]

[./Dirichlet1]
type = DirichletBC
boundary = 'left'
variable = T
value = 0.1
[../]

[./Dirichlet2]
type = DirichletBC
boundary = 'right'
variable = T
value = 1.5
[../]

[]

[Postprocessors]
[./dt]
type = TimestepSize
[../]

[]

[Preconditioning]
active = ' '
[./SMP]
type = SMP
full = true
petsc_options_iname = '-ksp_gmres_restart -snes_atol -snes_rtol -ksp_rtol -pc_type -build_twosided'
petsc_options_value = ' 160 1e-10 1e-8 1e-6 bjacobi allreduce'
[../]
[]

[Executioner]
type = Transient
scheme = BDF2
solve_type = newton

petsc_options_iname = '-pc_type -ksp_grmres_restart -sub_ksp_type -sub_pc_type -pc_asm_overlap'
petsc_options_value = 'asm 31 preonly lu 1'
l_max_its = 30
nl_abs_tol = 1e-7
l_tol = 1e-7
nl_max_its = 30
nl_rel_tol = 1e-7
end_time = 30000000
dt=0.50

[]

[Outputs]
execute_on = 'timestep_end'
exodus = true
interval=2
[]

**Comment by GiudGiud** [comment]
You should have an exodus file output. Look in the folder where you ran the input, with a .e file extension

**Reply by rakeshm22** [reply]
Yes, it generated an output file with a .e extension, but when I tried to plot it in ParaView, I encountered the following error.
Please find attachment.
![Screenshot from 2024-12-31 15-38-55](https://github.com/user-attachments/assets/6d036621-8a83-4f31-b1ed-1d80da9adbda)


Could you please let me know the correct flow for defining auxiliary variables, kernels, periodic boundary conditions, and the free energy? Did I define them correctly.

**Reply by GiudGiud** [reply]
Ive never seen this before. It looks like you have a field output twice. Can you remove it from the material property output?
